glusterd2/volume/struct.go

```diff
@@ -46,6 +46,8 @@ const (
 	VkeyScrubFrequency = "bit-rot.scrub-freq"
 	// VkeyScrubThrottle is the key for controls scrubber throttle
 	VkeyScrubThrottle = "bit-rot.scrub-throttle"
+	//VkeyFeaturesGfproxyd is the key which enables or disables gfproxyd
+	VkeyFeaturesGfproxyd = "features.gfproxyd"
```

Comment:
I don't know when this started, but features shouldn't be modifying core packages to add their stuff. Ideally, to add these sorts of keys, the core package should provide some sort of interface that plugins use to add things they want.

As I see, currently this isn't used else where out of the gfproxyd plugin. So you can just move it to the plugin.

Reply:
Volfile generation may use it. This is common problem for many features. For example bitrot. Unless we provide a way to store enabled features this we can't avoid.

I asked the same question here https://github.com/gluster/glusterd2/pull/413#discussion_r167149739

I feel we should allow this till we implement infra to store this info.

Reply:
As @kshlm suggested, move this to `plugins/gfproxyd`